### PR TITLE
Test deactivation controller with UserTiers

### PR DESCRIPTION
**Pull request by rajivnathan**
Related PR: https://github.com/codeready-toolchain/host-operator/pull/637

- tests that user deactivation works with both UserTier and NSTemplateTier (NSTemplateTier deactivation test will be removed after UserTier migration is complete)
- updates Verify functions to accept both `userTierName` and `spaceTierName` where `userTierName` corresponds to the expected MUR TierName and `spaceTierName` corresponds to  the expected Space TierName
- split `MoveUserToTier` helper function into `MoveSpaceToTier` and `MoveMURToTier`

### CODE REVIEW

File: test/e2e/user_management_test.go [hunk 1]
```diff
@@ -115,8 +115,11 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
 			Execute().Resources()
 
+		// TODO remove once UserTier migration is completed
+		s.promoteToDefaultUserTier(hostAwait.Client, mur1)
+
```
Comment (MatousJobanek) [inline test/e2e/user_management_test.go:120]: what is the purpose of promoting the user when it is deactivated manually?

File: test/e2e/user_management_test.go [hunk 2]
```diff
@@ -125,6 +128,9 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
 			Execute().Resources()
 
+		// TODO remove once UserTier migration is completed
+		s.promoteToDefaultUserTier(hostAwait.Client, mur2)
+
```
Comment (MatousJobanek) [inline test/e2e/user_management_test.go:133]: the same here and also in some other cases

File: test/e2e/user_management_test.go [hunk 3]
```diff
@@ -170,15 +179,18 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
 			Execute().Resources()
 
-		// Get the base tier that has deactivation disabled
-		baseDeactivationDisabledTier, err := hostAwait.WaitForNSTemplateTier("basedeactivationdisabled")
+		// TODO remove once UserTier migration is completed
+		s.promoteToDefaultUserTier(hostAwait.Client, murMember1)
```
Comment (MatousJobanek) [inline test/e2e/user_management_test.go:183]: we don't need that when the user will be promoted to `nodeactivation` tier in the following lines, right?

**Review by MatousJobanek — COMMENTED** [review]
TBH, it's a bit hard to follow all the changes in the PR since some of them are copy-paste with some additional changes :man_shrugging: 
BTW, we also need to modify the NSTemplateTier checks that verify the deactivation time - are you going to do that in the next PR?

File: testsupport/tiers/tier_setup.go [hunk 4]
```diff
@@ -156,12 +149,25 @@ func duplicateTierTemplate(hostAwait *HostAwaitility, namespace, tierName, origT
...
+
+func MoveMURToTier(t *testing.T, hostAwait *HostAwaitility, username, tierName string) {
+	t.Logf("moving masteruserrecord '%s' to user tier '%s'", username, tierName)
+	mur, err := hostAwait.WaitForMasterUserRecord(username)
 	require.NoError(t, err)
-	_, err = hostAwait.WaitForChangeTierRequest(changeTierRequest.Name, toBeComplete)
+
+	mur.Spec.TierName = tierName
+
+	err = hostAwait.Client.Update(context.TODO(), mur)
```
Comment (MatousJobanek) [inline testsupport/tiers/tier_setup.go:170]: the same
```suggestion
	_, err = hostAwait.UpdateMasterUserRecord(false, mur.Name, func(mur *toolchainv1alpha1.MasterUserRecord) {
		mur.Spec.TierName = tierName
	})
```

File: test/e2e/user_management_test.go (hunk 1 above)
Comment (rajivnathan) [reply test/e2e/user_management_test.go]: good point, it's not needed. Removed this and other cases in a6cd3440370feefacd130d319af1cd0fc90c9990

File: test/e2e/user_management_test.go (hunk 2 above)
Comment (rajivnathan) [reply test/e2e/user_management_test.go]: Removed this and other cases in a6cd3440370feefacd130d319af1cd0fc90c9990

File: test/e2e/user_management_test.go (hunk 3 above)
Comment (rajivnathan) [reply test/e2e/user_management_test.go:174]: Removed this and other cases in a6cd3440370feefacd130d319af1cd0fc90c9990

Comment (rajivnathan) [reply test/e2e/user_management_test.go]: Right again! Removed it c6022abad03a905ac5e3691007be3e603a190b62

File: testsupport/tiers/tier_setup.go [hunk 5]
```diff
@@ -156,12 +149,25 @@ func duplicateTierTemplate(hostAwait *HostAwaitility, namespace, tierName, origT
...
-	err := hostAwait.CreateWithCleanup(context.TODO(), changeTierRequest)
+func MoveSpaceToTier(t *testing.T, hostAwait *HostAwaitility, spacename, tierName string) {
+	t.Logf("moving space '%s' to space tier '%s'", spacename, tierName)
+	space, err := hostAwait.WaitForSpace(spacename)
+	require.NoError(t, err)
+
+	space.Spec.TierName = tierName
+
+	err = hostAwait.Client.Update(context.TODO(), space)
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (rajivnathan) [reply testsupport/tiers/tier_setup.go]: I have updated https://github.com/codeready-toolchain/host-operator/pull/637 so that the space controller will remove the old tier hash label as part of the space update. Also, the deactivation controller now resets the Deactivating state so that when a user is promoted they will still receive the Deactivating notification at the correct time so this fixes the e2e test failure for this PR. This also means we don't need to use ChangeTierRequest for tier updates anymore, can promote MURs and Spaces individually.

**Review by MatousJobanek — APPROVED** [review]
Looks good, but don't forget to fix the tier checks in the following PR (as I mentioned in my previous review comment)
https://github.com/codeready-toolchain/toolchain-e2e/pull/550#pullrequestreview-959068631

> we also need to modify the NSTemplateTier checks that verify the deactivation time - are you going to do that in the next PR?
